api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -73,4 +73,13 @@ RewriteFiles rewriteFiles(Set<DataFile> dataFilesToReplace, Set<DeleteFile> dele
    * @return this for method chaining
    */
   RewriteFiles validateFromSnapshot(long snapshotId);
+
+  /**
+   * Use the specified sequence number for the new manifest of the data files added in this update,
+   * instead of inheriting the sequence number of the snapshot that will be created.
+   *
+   * @param sequenceNumber a sequence number
+   * @return this for method chaining
+   */
+  RewriteFiles overrideSequenceNumberForNewDataFiles(long sequenceNumber);
```

Comment:
I think there is probably a shorter, more descriptive name for this. Something like `commitAtSequenceNumber`?

Reply:
The reasons I used this long name are:
1. this is not actually committing with the given sequence number, the overall sequence number still increments, but the sequence number provided is used in the manifest file of the newly added data files. Please let me know if this is the desired effect or not
2. I was thinking maybe we will need to do something similar for delete files, that's why there is a suffix of `ForNewDataFiles`.

Reply:
Good points about naming, but this still seems long to me. Maybe we should pass the sequence number in with the data files then?

We could add `rewriteFiles(Iterable<DataFile> toRemove, Iterable<DataFile> toAdd, long sequenceNumber)`. I think that's a bit more clear because you don't remove files at a sequence number. We also don't have the problem that it is confusing for delete files because it is the data file method only.

Reply:
This looks like a fairly clean way to configure everything now that I'm looking at context. Up to you whether to use `rewriteFiles` to pass it or a new method, but I think we should come up with a shorter name if we go with the new method approach. What about `dataFileSequenceNumber`?

Reply:
yes I think the new rewriteFiles method sounds good to me, let me update with that.